Prox/Prox/Models/PlaceProviders.swift

```diff
@@ -0,0 +1,159 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+
+protocol PlaceProvider {
+    var id: String? { get }
+    var name: String? { get}
+    var description: String? { get }
+    var categories: (names: [String], ids: [String]) { get }
+    var latLong: CLLocationCoordinate2D? { get }
+    var photoURLs: [URL] { get }
+    var url: URL? { get }
+    var address: String? { get }
+    var hours: OpenHours? { get }
+    var rating: Float? { get }
+    var totalReviewCount: Int? { get }
+}
+
+class SourcePlaceProvider: PlaceProvider {
```

Comment:
nit: `SourcePlaceProvider` is a confusing name to me – I was thinking "it's the source of everything" (singular) rather than "here is one of our sources, like Yelp".

I think the implementation should really be called, `PlaceProvider` but the interface gets in the way. We could do an ugly: `IPlaceProvider` or `PlaceProviderProtocol` (I'm not a fan of `*Impl`). Or maybe the implementation could be called `SinglePlaceProvider` in contrast to `CompositePlaceProvider`?

Then again, xcode makes refactoring bullshit so maybe just leave it as is. :)